docs/basics/101-146-providers.rst

```diff
@@ -0,0 +1,205 @@
+.. _providers:
+
+Configure custom data access
+----------------------------
+
+DataLad can download files via the ``http``, ``https``, ``ftp``, and ``s3``
+protocol from various data storage solutions via its downloading commands
+(:command:`datalad download-url`, :command:`datalad addurls`,
+:command:`datalad get`).
+If data retrieval from a storage solution requires *authentication*,
+for example via a username and password combination, DataLad provides an
+interface to query, request, and store the most common type of credentials that
+are necessary to authenticate, for a range of authentication types.
+There are a number of natively supported types of authentication and out-of-the
+box access to a broad range of access providers, from common solutions such as
+`S3 <https://aws.amazon.com/s3/?nc1=h_ls>`_ to special purpose solutions, such as
+`LORIS <https://loris.ca/>`_. However, beyond natively supported services,
+custom data access can be configured as long as the required authentication
+and credential type are supported. This makes DataLad even more flexible for
+retrieving data.
+
+Basic process
+^^^^^^^^^^^^^
+
+For any supported access type that requires
+authentication, the procedure is always the same:
+Upon first access via any downloading command, users will be prompted for their
+credentials from the command line. Subsequent downloads handle authentication
+in the background as long as the credentials stay valid. An example of this
+credential management is shown in the usecase :ref:`usecase_HCP_dataset`:
+Data is stored in S3 buckets that require authentication with AWS credentials.
+The first :command:`datalad get` to retrieve any of the data will prompt for
+the credentials from the terminal. If the given credentials are valid, the
+requested data will be downloaded, and all subsequent retrievals via
+:command:`get` will authenticate automatically, without user input, as long as
+the entered credentials stay valid.
+
```

Comment:
It would be good to add a pointer to https://keyring.readthedocs.io which is what we use for credential storage. It essentially interfaces with any built-in credential store of the major OSes, but in addition has a bunch of special-purpose backends that come handy in corner cases (e.g. HPC/cluster computing, where no interactive sessions are available). This makes the whole system really flexible.

Reply:
Thanks for this pointer. Yes, I agree. Will add.